frontend/packages/client/src/components/UploadImageModal.js

```diff
@@ -0,0 +1,325 @@
+import React, { useCallback, useState, useEffect, useMemo } from "react";
+import { useDropzone } from "react-dropzone";
+import { useFileUploader } from "../hooks";
+import { Upload, Bin } from "../components/Svg";
+import { Loader } from "../components";
+
+const IMAGE_STATUS = {
+  notStarted: "not-started",
+  uploading: "uploading",
+  uploaded: "uploaded",
+};
+
+function ImageUploader({
```

Comment:
In general, I would favor splitting components into separate files and using folders to group them by functionality/larger components, with the top-level `components` folder being mainly for those which are shared across multiple areas. What do you think?

Reply:
yes, I agree. We should take this approach...perhaps there might be some refactor to be done to achieve this on all the repo but I think it's convenient effort